hcipy/optics/aberration.py

```diff
@@ -51,6 +52,49 @@ def psd(grid):
 
     return Field(screen * aperture, pupil_grid)
 
+def make_high_pass_power_law_error(pupil_grid, std, diameter, cutoff_frequency, exponent=-2.5, aperture=None, filter_shape_parameter=15):
+    '''Create an error surface from a high-pass filtered power-law power spectral density.
+
+    Parameters
+    ----------
+    pupil_grid : Grid
+        The grid on which to calculate the error.
+    std : scalar
+        The standard deviation of the wavefront aberration in meters.
+    diameter : scalar
+        The diameter over which the ptv is calculated.
+    cutoff_frequency : scalar
+        The cutoff frequency of the high-pass filter.
+    exponent : scalar
+        The exponent of the power law.
+    aperture : Field
+        The mask over which to calculate the ptv. A circular aperture with diameter
+        `diameter` is used if this is not given.
+    filter_shape_parameter : scalar
+        The shape parameter for the general Gaussian high-pass filter fuction. Default value is 15.0
+
+    Returns
+    -------
+    Field
+        The surface error calculated on `pupil_grid`.
+    '''
+    def filter_function(fourier_grid):
+        return 1 - make_general_gaussian_aperture(2 * cutoff_frequency, filter_shape_parameter)(fourier_grid)
+
+    ff = FourierFilter(pupil_grid, filter_function, q=2)
+
+    if aperture is None:
+        aperture = make_circular_aperture(diameter)(pupil_grid)
+    aperture_mask = aperture > 0
+
+    phase_screen = make_power_law_error(pupil_grid, 1.0, diameter, exponent=exponent, aperture=None)
+    phase_screen = np.real(ff.forward(phase_screen + 0j))
+
+    phase_screen_rms = np.std(phase_screen[aperture_mask > 0]) if aperture is not None else np.std(phase_screen)
+    phase_screen *= std / phase_screen_rms
```

Comment:
I don't really like this style of normalization. You could also calculate the expected RMS for a specific aperture and normalize by that. But, this is the question of: when you use the word RMS, do you mean the RMS on the given aperture or the expected RMS on that aperture. My fear is always that you'll bias your phase screens statistically if you normalize the way this is currently written.

I'm fine with this, but I wanted your opinions on this.

Reply:
I understand what you mean. I usually want a specific variance and then I vary that variance myself to explore the range (or I stochastically pick a variance). I don't see the difference between normalizing by std in this way or what we currently do by normalizing on peak to valley. I can also normalize it on peak to valley to make it more similar to the other function.